app/controllers/twitter.js

```diff
@@ -1,23 +1,211 @@
+		"name": "Utah",
+		"full_name": "Utah, USA",
+		"country_code": "US",
+		"country": "United States",
+		"bounding_box": {
+			"type": "Polygon",
+			"coordinates": [
+				[
+					[-114.052999, 36.997905],
+					[-114.052999, 42.001619],
+					[-109.041059, 42.001619],
+					[-109.041059, 36.997905]
+				]
+			]
+		},
+		"attributes": {}
+	},
+	"contributors": null,
+	"is_quote_status": false,
+	"retweet_count": 0,
+	"favorite_count": 0,
+	"entities": {
+		"hashtags": [{
+			"text": "SLC",
+			"indices": [74, 78]
+		}, {
+			"text": "PokemonGO",
+			"indices": [79, 89]
+		}],
+		"urls": [{
+			"url": "https://t.co/JWpOPLfVpA",
+			"expanded_url": "https://maps.google.com/maps?q=40.768927,-111.808184",
+			"display_url": "maps.google.com/maps?q=40.7689…",
+			"indices": [90, 113]
+		}],
+		"user_mentions": [],
+		"symbols": []
+	},
+	"favorited": false,
+	"retweeted": false,
+	"possibly_sensitive": false,
+	"filter_level": "low",
+	"lang": "en",
+	"timestamp_ms": "1471551440854"
+}
+
+important fields: 
+	create_at	: UTC time when this Tweet was created.
+	coordinates	: he longitude and latitude of the Tweet’s location.
+	text		: The actual UTF-8 text of the status update.T.
 
-let Writable = require('stream').Writable,
+*/
+
+let fs = require('fs'),
+    config = require('../../config'),
+	Writable = require('stream').Writable,
     TwitterStream = require('twitter-stream-api'),
-    config = require('../../config');
+	pokemonTweet = require(__appbase + 'models/pokemonAppearances.js'),
+    pokemonListPath = require(__appbase + '../resources/json/pokemonIdAndName.json');
 
 const Twitter = new TwitterStream(config.twitter);
 
+/*
+	keywords used to search for in the tweets
+	,  : represents OR
+	' ': space represents AND 
+*/
+const tweetSearchKeywords = 'pokemon catch,pokemon saw,pokemon attack,pokemon find,pokemon caught,pokemon attacked,pokemon found,pokemon appeared,#foundPokemon,#caughtPokemon,#pokemongo,a wild appeared until #pokemongo';
+
+// array of names of pokemons
+var pokemonNameList = [];
+
+// process the tweet object
 let Output = Writable({objectMode: true});
 Output._write = function (obj, enc, next) {
-    // the tweet json
-    console.log(JSON.stringify(obj));
-    // the tweet id and text
-    console.log(obj.id, obj.text);
+	checkIfTweetHasPokemonName(obj);
     next();
 };
 
+/*
+	check if tweet object has pokemon term
+ */
+function checkIfTweetHasPokemonName(tweetObj) {
+	// check if the tweet text contains the important keywords
+	if (tweetObj.text.toLowerCase().indexOf('caught') != -1 || 
+		tweetObj.text.toLowerCase().indexOf('saw') != -1 || 
+		tweetObj.text.toLowerCase().indexOf('attacked') != -1 ||
+		tweetObj.text.toLowerCase().indexOf('found') != -1  ||
+		tweetObj.text.toLowerCase().indexOf('appeared') != -1 ) {
+
+		// for each pokemon name check if it is inside the tweet text
+		for(var i = 0; i < pokemonNameList.length; i++) {
+			if (tweetObj.text.toLowerCase().indexOf(pokemonNameList[i]) != -1){
```

Comment:
pokemonNameList.lengt could also be stored in a variable instead of accessing the object property in the loop each time.